lib/string_reverse.rb

```diff
@@ -1,6 +1,18 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: ? O(n), n is the number of characters in the string.
+# Space complexity: ? O(1)
 def string_reverse(my_string)
-  raise NotImplementedError
+  if my_string == nil
+    return nil
+  end
+  i = 0

```

Comment:
Tiny (personal preference) nit: some people (and companies) prefer more explanatory (but less concise) variable names than `i` and `j`.